go-controller/pkg/ovn/obj_retry.go

```diff
@@ -1191,7 +1306,21 @@ func (oc *Controller) resourceRetry(r *retryObjs, lre *localRetryEntry, now time
 	klog.Infof("Retry object setup: %s %s", r.oType, objKey)
 
 	if entry.newObj != nil {
-		entry.newObj = lre.kObj
+		// get the latest version of the object from the informer;
+		// if it doesn't exist we are not going to create the new object.
+		kObj, err := oc.getResourceFromInformerCache(r.oType, objKey)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				klog.Infof("%s %s not found in the informers cache,"+
+					" not going to retry object create", r.oType, objKey)
+				kObj = nil
+			} else {
+				klog.Errorf("Failed to look up %s %s in the informers cache,"+
+					" will retry later: %v", r.oType, objKey, err)
+				return
+			}
+		}
+		entry.newObj = kObj
```

Comment:
This change seems unrelated, can you split it to a different commit and explain?

Reply:
It is related - this part was modes from `iterateRetryResources` https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/obj_retry.go#L1263-L1276 to reduce the need for additional entry lock - now it is a part of `resourceRetry`